pkg/controller/metrics/provision_underway_collector.go

```diff
@@ -76,6 +82,8 @@ func (cc provisioningUnderwayCollector) Collect(ch chan<- prometheus.Metric) {
 			GetClusterDeploymentType(&cd),
 			condition,
 			reason,
+			platform,
```

Comment:
We are hitting the [number 10](https://prometheus.io/docs/practices/instrumentation/#do-not-overuse-labels) here - just a caution to keep in mind

Reply:
Fair point. This is probably abusing prometheus because we're not well positioned to run some kind of script on an alert that would gather this info for us, or a link to a page we could view for info. Platform probably is fine here as someone might want to route alerts depending on that. Image set probably is not great here... 

Do you think it's safe for now? 

We should definitely think about how we could improve in future.

Reply:
It is just a caution and/or a good practice. I think the alert for a failing nightly would help - we can also try and not log a metric if not available - since we don't have to bother cleaning up (for ex, say imageset is none, then we don't need prometheus to scrape the none, just don't report it)
Regardless, I think we are okay for now, we can revisit alternatives if we want to add more labels in the future